vault/resource_auth_backend.go

```diff
@@ -171,38 +189,44 @@ func authBackendUpdate(d *schema.ResourceData, meta interface{}) error {
 	if !d.IsNewResource() {
 		path, e = util.Remount(d, client, consts.FieldPath, true)
 		if e != nil {
-			return e
+			return diag.FromErr(e)
 		}
 	}
 
-	backendType := d.Get("type").(string)
-	var input api.MountConfigInput
+	backendType := d.Get(consts.FieldType).(string)
+	var config api.MountConfigInput
 	var callTune bool
 
-	if d.HasChange("tune") {
+	if d.HasChange(consts.FieldTune) {
 		log.Printf("[INFO] Auth '%q' tune configuration changed", path)
 
-		if raw, ok := d.GetOk("tune"); ok {
+		if raw, ok := d.GetOk(consts.FieldTune); ok {
 			log.Printf("[DEBUG] Writing %s auth tune to '%q'", backendType, path)
 
-			input = expandAuthMethodTune(raw.(*schema.Set).List())
+			config = expandAuthMethodTune(raw.(*schema.Set).List())
 		}
 		callTune = true
 	}
 
-	if d.HasChange("description") && !d.IsNewResource() {
-		desc := d.Get("description").(string)
-		input.Description = &desc
+	if d.HasChanges(consts.FieldIdentityTokenKey, consts.FieldDescription) && !d.IsNewResource() {
+		desc := d.Get(consts.FieldDescription).(string)
+		config.Description = &desc
+
+		useAPIVer117Ent := provider.IsAPISupported(meta, provider.VaultVersion117) && provider.IsEnterpriseSupported(meta)
+		if useAPIVer117Ent {
+			config.IdentityTokenKey = d.Get(consts.FieldIdentityTokenKey).(string)
+		}
+
 		callTune = true
 	}
 
 	if callTune {
-		if err := tuneMount(client, "auth/"+path, input); err != nil {
-			return err
+		if err := tuneMount(client, "auth/"+path, config); err != nil {
```

Comment:
Similar to GCP, we should update this to pass in the context. I can update these in the follow up PR if that works with folks.

Reply:
Works for me! 👍🏼